### ISSUE TITLE
Ensure we're properly handling "request body stream exhausted" errors

### ISSUE THREAD

**Issue by aerych** [wordpress-mobile/WordPress-iOS#3364]
We've had at least one report in the forums of this error occurring when uploading a featured image. 
Mattt has a recommended approach for detaling with it in his reply to this SO thread: https://stackoverflow.com/a/16762176
Let's double check our process and retry a failed upload when this error occurs on slower connections.

**Comment by bummytime** [comment]
I've spend a bit of time trying to recreate this error via media library uploads, Aztec media uploads, etc. I've tried various network link conditioner settings and even drove out to a location where I could test on a very spotty T-Mobile network (where I was able to consistently recreate this "request body steam exhausted" issue [in the past](https://github.com/wordpress-mobile/WordPress-Editor-iOS/issues/74)).

I am beginning to think this may not be issue any more. Reasons for this may include:
* The async media efforts
* Cell network infrastructure improvements
* iOS/Foundation framework improvements
* ???

Given that we have not heard much about this issue since 2015 and the media async work exists now (which includes retry), I am going to close this issue.

/cc @SergioEstevao @frosty
